cpp/include/cudf/detail/scatter.cuh

```diff
@@ -115,6 +115,21 @@ struct column_scatterer_impl<string_view, MapIterator> {
   }
 };
 
+template <typename MapIterator>
```

Comment:
This section of code was moved above the `column_scatter_impl` for dictionary since it needs to use it for scattering the indices.